File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        // unordered_set<int>seen;
        // for(const int i:nums)
        // {
        //     if (!seen.insert(i).second)
        //     {
        //         return true;
        //     }
        // }

        sort(nums.begin(),nums.end());
        for (int i=1;i<nums.size();i++)
        {
            if (nums[i]==nums[i-1])
            {
                return true;
            }
        }
       return false;

    }
};

File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        Map<Integer, Integer>fre = new HashMap<>();

        for (int i:nums)
        {
            fre.put(i,fre.getOrDefault(i,0)+1);
        }
        
        for (Integer value : fre.values())
        {
            if (value>=2)
            {
                return true;
            }
        }
        return false;
        
    }
}